h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeNode.java

```diff
@@ -267,6 +269,33 @@ public void print(PrintStream out, String description) {
     out.println("            rightChild:  " + ((rightChild != null) ? rightChild.getName() : ""));
   }
 
+  public void log(LoggingLevel loggingLevel, String description) {
```

Comment:
this is a relatively expensive function - perhaps the message shouldn't be constructed if won't be logged in the end

Reply:
added default

Reply:
also I dont like the duplication of this code, we construct the same string as we do in print(), personally I would just have `print() { out.print(this.getPrintString(); }` and then getPrintString() method with the current contents of log() method, and in sharedtreemojoModel I would have 
`if (logger.isEnabled(ERROR) { logger.error(node.getPrintString()); }`